packages/react/src/auto/polaris/PolarisAutoFileInput.tsx

```diff
@@ -0,0 +1,175 @@
+import type { DropZoneProps } from "@shopify/polaris";
+import { Box, DropZone, InlineError, InlineStack, Thumbnail } from "@shopify/polaris";
+import { NoteIcon } from "@shopify/polaris-icons";
+import { filesize } from "filesize";
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import type { Control } from "react-hook-form";
+import { useController } from "react-hook-form";
+import { useApi } from "../../GadgetProvider.js";
+import type { GadgetOnlyImageFileFieldValidation, GadgetRangeFieldValidation } from "../../internal/gql/graphql.js";
+import { isAutoFileFieldValue, type AutoFileFieldValue } from "../../validationSchema.js";
+import { useFieldMetadata } from "../hooks/useFieldMetadata.js";
+
+const imageFileTypes = ["image/jpeg", "image/png", "image/svg+xml", "image/webp", "image/gif"];
+
+export const PolarisAutoFileInput = (props: { field: string; control?: Control<any> } & DropZoneProps) => {
+  const [openFileDialog, setOpenFileDialog] = useState(false);
+  const { field: fieldApiIdentifier, control, ...rest } = props;
+  // For showing a preview of the file
+  const [imageThumbnailURL, setImageThumbnailURL] = useState<string>();
+  const { path, metadata } = useFieldMetadata(fieldApiIdentifier);
+  const api = useApi();
+
+  const {
+    field: fieldProps,
+    fieldState: { error },
+  } = useController({
+    control,
+    name: path,
+  });
+
+  const rangeValidation = useMemo(
+    () => metadata.configuration.validations.find((validation) => validation?.__typename === "GadgetRangeFieldValidation"),
+    [metadata]
+  ) as GadgetRangeFieldValidation | undefined;
+
+  useEffect(() => {
+    if (fieldProps.value && isAutoFileFieldValue(fieldProps.value) && fieldProps.value.url) {
+      setImageThumbnailURL(fieldProps.value.url);
+    }
+  }, [fieldProps.value]);
+
+  const onlyImages = useMemo(() => {
+    const imageValidation = metadata.configuration.validations.find(
+      (validation) => validation?.__typename === "GadgetOnlyImageFileFieldValidation"
+    ) as GadgetOnlyImageFileFieldValidation | undefined;
+    if (!imageValidation) return;
+
+    const acceptedTypes = [...imageFileTypes];
+    if (imageValidation.allowAnimatedImages) {
+      acceptedTypes.push("image/gif");
+    }
+
+    return { acceptedTypes, allowAnimatedImages: imageValidation.allowAnimatedImages };
+  }, [metadata]);
+
+  const handleDropZoneDrop = useCallback(
+    async (_dropFiles: File[], acceptedFiles: File[], _rejectedFiles: File[]) => {
+      const file = acceptedFiles[0];
+      const baseFileValue = {
+        fileName: file.name,
+        mimeType: file.type,
+      } satisfies AutoFileFieldValue;
+
+      setImageThumbnailURL(window.URL.createObjectURL(file));
+
+      if (
+        rangeValidation &&
+        ((rangeValidation.min && file.size < rangeValidation.min) || (rangeValidation.max && file.size > rangeValidation.max))
```

Comment:
Also seems to overlap with similar validation in `packages/react/src/validationSchema.tsx`